coriolis/tests/test_keystone.py

```diff
@@ -0,0 +1,296 @@
+            impersonation=True,
+            role_names=self.ctxt.roles)
+        self.assertEqual(self.ctxt.trust_id, self.trust.id)
+
+    def test_create_trust_with_existing_trust_id(self):
+            self.ctxt.trust_id = 'test_trust_id'
+            result = keystone.create_trust(self.ctxt)
+
+            self.assertEqual(result, None)
+            self.assertEqual(self.ctxt.trust_id, 'test_trust_id')
+
+    @mock.patch('coriolis.keystone._get_trusts_auth_plugin')
+    @mock.patch.object(keystone.kc_v3, 'Client')
+    @mock.patch.object(keystone.loading, 'get_plugin_loader')
+    @mock.patch.object(keystone.ks_session, 'Session')
+    def test_create_trust_unauthorized_exception(self,
+                                                 mock_session,
+                                                 mock_get_plugin_loader,
+                                                 mock_client,
+                                                 mock_get_trusts_auth_plugin):
+        mock_get_trusts_auth_plugin.return_value = self.trusts_auth_plugin
+        mock_get_plugin_loader.return_value.\
+            load_from_options.return_value = self.auth
+        mock_session.return_value = self.session
+        mock_client.return_value = self.client
+        self.trusts_auth_plugin.get_user_id.side_effect = \
+            ks_exceptions.Unauthorized
+
+        self.assertRaises(exception.NotAuthorized, keystone.create_trust,
+                          self.ctxt)
+
+    @mock.patch('coriolis.keystone._get_trusts_auth_plugin')
+    @mock.patch.object(keystone.kc_v3, 'Client')
+    @mock.patch.object(keystone.ks_session, 'Session')
+    def test_delete_trust(self, mock_session, mock_client,
+                          mock_get_trusts_auth_plugin):
+        mock_get_trusts_auth_plugin.return_value = self.trusts_auth_plugin
+        self.ctxt.trust_id = 'test_trust_id'
+        mock_session.return_value = self.session
+        mock_client.return_value = self.client
+
+        keystone.delete_trust(self.ctxt)
+
+        mock_get_trusts_auth_plugin.assert_called_once_with('test_trust_id')
+        mock_session.assert_called_once_with(
+            auth=self.trusts_auth_plugin, verify=True)
+        mock_client.assert_called_once_with(session=self.session)
+        self.client.trusts.delete.assert_called_once_with('test_trust_id')
+        self.assertEqual(self.ctxt.trust_id, None)
+
+    @mock.patch('coriolis.keystone._get_trusts_auth_plugin')
+    @mock.patch.object(keystone.kc_v3, 'Client')
+    @mock.patch.object(keystone.ks_session, 'Session')
+    def test_delete_trust_with_not_found_exception(
+        self, mock_session, mock_client, mock_get_trusts_auth_plugin):
+        mock_get_trusts_auth_plugin.return_value = self.trusts_auth_plugin
+        self.ctxt.trust_id = 'test_trust_id'
+        mock_session.return_value = self.session
+        mock_client.return_value = self.client
+        self.client.trusts.delete.side_effect = ks_exceptions.NotFound
```

Comment:
NIT: Please make this look better. This throws `E125` warning because it's the same indent as the method code below. 
Acceptable style:
```suggestion
    def test_delete_trust_with_not_found_exception(
            self, mock_session, mock_client, mock_get_trusts_auth_plugin):
        mock_get_trusts_auth_plugin.return_value = self.trusts_auth_plugin
        self.ctxt.trust_id = 'test_trust_id'
        mock_session.return_value = self.session
        mock_client.return_value = self.client
        self.client.trusts.delete.side_effect = ks_exceptions.NotFound
```
or 
```suggestion
    def test_delete_trust_with_not_found_exception(
        self, mock_session, mock_client, mock_get_trusts_auth_plugin
    ):
        mock_get_trusts_auth_plugin.return_value = self.trusts_auth_plugin
        self.ctxt.trust_id = 'test_trust_id'
        mock_session.return_value = self.session
        mock_client.return_value = self.client
        self.client.trusts.delete.side_effect = ks_exceptions.NotFound
```

Reply:
You're right, I missed it. Thanks!